src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessor.java

```diff
@@ -69,17 +75,36 @@ private boolean isReady(BlobItem blob, String containerName) {
     }
 
     private void processBlob(BlobClient blob) {
-        envelopeService
-            .findLastEnvelope(blob.getBlobName(), blob.getContainerName())
+        leaseAcquirer
+            .acquireFor(blob)
```

Comment:
It would be perfect if acquiring the lease and releasing it was done by the same class. It could be leaseAcquirer itself.

Reply:
fixed